elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/scaladsl/ElasticsearchFlow.scala

```diff
@@ -160,15 +158,62 @@ object ElasticsearchFlow {
                                     typeName: String,
                                     settings: ElasticsearchWriteSettings,
                                     elasticsearchClient: RestClient,
-                                    writer: MessageWriter[T]) =
-    Flow
-      .fromGraph(
-        new impl.ElasticsearchSimpleFlowStage[T, C](indexName, typeName, elasticsearchClient, settings, writer)
+                                    writer: MessageWriter[T]) = {
+    val (minBackoff, maxBackoff, maxRetries) = settings.retryLogic match {
+      case RetryNever => (Duration.Zero, Duration.Zero, 0)
+      case RetryAtFixedRate(retries, interval) => (interval, interval, retries)
+      case _ => throw new IllegalArgumentException("Unsupported retry logic")
+    }
+
+    def retryLogic(
+        results: immutable.Seq[WriteResult[T, (Int, C)]]
+    ): Option[(immutable.Seq[WriteMessage[T, (Int, C)]], immutable.Seq[WriteResult[T, (Int, C)]])] = {
+      val (successful, failed) = results.partition(_.success)
+
+      failed match {
+        case Nil => None
+        case failedResults => Some(failedResults.map(_.message) -> successful)
+      }
+    }
+
+    Flow[immutable.Seq[WriteMessage[T, C]]]
+      .map { messages =>
+        val indexedMessages = messages.zipWithIndex.map {
+          case (m, idx) =>
+            m.withPassThrough(idx -> m.passThrough)
+        }
+        indexedMessages -> Nil
+      }
+      .via(
+        Flow.fromGraph {
+          val retryFlowCoordinator =
+            BidiFlow.fromGraph(
+              new RetryFlowCoordinator[(immutable.Seq[WriteMessage[T, (Int, C)]],
+                                        immutable.Seq[WriteResult[T, (Int, C)]]),
+                                       immutable.Seq[WriteResult[T, (Int, C)]]](
+                minBackoff,
+                maxBackoff,
+                0,
+                maxRetries,
+                (_, results) => retryLogic(results)
+              )
+            )
+          val elasticsearchFlow = new impl.ElasticsearchSimpleFlowStage[T, (Int, C)](indexName,
+                                                                                     typeName,
+                                                                                     elasticsearchClient,
+                                                                                     settings,
+                                                                                     writer)
+          retryFlowCoordinator.joinMat(elasticsearchFlow)(Keep.right)
```

Comment:
Copy over the `RetryFlow.withBackoff` method so the switch to the Akka 2.6 implementation is even simpler.

Reply:
👍